### PR TITLE
#219, add a project under samples for atrium + spek

### PR DESCRIPTION
**Pull request by iamshubh**
----
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: samples/jvm/spek/src/test/kotlin/SmokeTest.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+        it("division sanity") {
+            expect {
+                // the below statement throws exception
+                calculator.divide(12, 0)
+            }.toThrow<IllegalArgumentException>()
+        }
+
+        it("defaults to fancy"){
+            expect(calculator.isFancy)
```
[the remark this answers is not in the document]
Comment (robstoll) [reply samples/jvm/spek/src/test/kotlin/SmokeTest.kt]: You need to change the code to resolve this issue :grin:

**Comment by robstoll** [comment]
@iamshubh sorry, I did not see you requested my review. One point is still open, please apply my suggestion (can be done directly via GitHub => click `commit Suggestion`) and I can merge this PR

**Comment by robstoll** [comment]
I tried to build your sample project and failed due to a missing gradle wrapper. Please add it (you can copy the /gradle/wrapper of Atrium

**Comment by robstoll** [comment]
I tried to build your sample project and failed due to a missing gradle wrapper. Please add it (you can copy the /gradle/wrapper of Atrium). Moreover, we want that the project gets build in CI as well (this way you would have noticed the failure yourself). For this:
- change .travis.yml => add another if (a second command):
  ```
  - |
    if [ $MODE == current ]; then
      ./samples/jvm/spek/gradlew -p ./samples/jvm/spek build       
    fi
- same same but different for the Windows build => change .github/workflows/java-windows.yml add another run
  ```

**Comment by iamshubh** [comment]
> I tried to build your sample project and failed due to a missing gradle wrapper. Please add it (you can copy the /gradle/wrapper of Atrium). Moreover, we want that the project gets build in CI as well (this way you would have noticed the failure yourself). For this:
> 
> * change .travis.yml => add another if (a second command):
>   ```
>   - |
>     if [ $MODE == current ]; then
>       ./samples/jvm/spek/gradlew -p ./samples/jvm/spek build       
>     fi
>   ```
> * same same but different for the Windows build => change .github/workflows/java-windows.yml add another run

Hey,

**Comment by iamshubh** [comment]
> * same same but different for the Windows build => change .github/workflows/java-windows.yml add another run

Added CI in .travis.yml.
could you add the ci configuration for windows

**Comment by robstoll** [comment]
What is the matter with Windows?

File: .travis.yml [hunk 2]
```diff
@@ -19,6 +19,7 @@ install: true
 script:
   - |
     if [ $MODE == current ]; then
+      ./samples/jvm/spek/gradlew -p ./samples/jvm/spek build
```
Comment (robstoll) [inline .travis.yml]: Please in an own if as I wrote. this way we would not see that it fails

File: samples/jvm/spek/build.gradle [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    id 'org.jetbrains.kotlin.jvm' version '1.3.41'
+}
+
+group 'com.atrium.spek'
+version '1.0-SNAPSHOT'
+
+apply plugin: 'kotlin'
+
+repositories {
+    // atrium jvm installation: more info here -> https://www.kotlinresources.com/library/atrium/#jvm
```
Comment (robstoll) [inline samples/jvm/spek/build.gradle]: please refer to the README in github and not to kotlinresources

File: samples/jvm/spek/build.gradle [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    jcenter()
+
+    // dependencies for spek
+    mavenLocal()
+    // uncomment to use development build for <u>spek<u>
+    maven { url "https://dl.bintray.com/spekframework/spek-dev" }
```
Comment (robstoll) [inline samples/jvm/spek/build.gradle]: either change the comment or comment this out :)
```suggestion
    // maven { url "https://dl.bintray.com/spekframework/spek-dev" }
```

**Comment by robstoll** [comment]
@iamsubh shall I take over?

File: .travis.yml (hunk 2 above)
Comment (iamshubh) [reply .travis.yml]: didn't get what you mean?

**Comment by iamshubh** [comment]
yup, you please go ahead

**Comment by robstoll** [comment]
Can you re-open it then, I'll merge it into another branch

File: .travis.yml (hunk 2 above)
Comment (robstoll) [reply .travis.yml]: Only the last statement within an if counts towards the exit code of the whole if. So if the first statement fails, then the second is still carried out and if this one succeeds, then the whole if succeeds. Therefore, don't but the additional `gr` inside the existing if but duplicate the if.

**Pull request closed without merging**